src/CaptureFile/CaptureFileTest.cpp

```diff
@@ -20,15 +20,127 @@ static constexpr const char* kNotAnAnswerString = "Some odd number, not the answ
-  orbit_grpc_protos::ClientCaptureEvent event;
+using orbit_grpc_protos::ClientCaptureEvent;
+using testing::HasSubstr;
+
+static ClientCaptureEvent CreateInternedStringCaptureEvent(uint64_t key, const std::string& str) {
+  ClientCaptureEvent event;
   orbit_grpc_protos::InternedString* interned_string = event.mutable_interned_string();
   interned_string->set_key(key);
   interned_string->set_intern(str);
   return event;
 }
 
+TEST(CaptureFile, CreateCaptureFileAndReadMainSection) {
+  auto temporary_file_or_error = orbit_base::TemporaryFile::Create();
+  ASSERT_TRUE(temporary_file_or_error.has_value()) << temporary_file_or_error.error().message();
+  orbit_base::TemporaryFile temporary_file = std::move(temporary_file_or_error.value());
+
+  std::string temp_file_name = temporary_file.file_path().string();
+
+  auto output_stream_or_error = CaptureFileOutputStream::Create(temp_file_name);
+  ASSERT_TRUE(output_stream_or_error.has_value()) << output_stream_or_error.error().message();
+
+  std::unique_ptr<CaptureFileOutputStream> output_stream =
+      std::move(output_stream_or_error.value());
+
+  EXPECT_TRUE(output_stream->IsOpen());
+
+  orbit_grpc_protos::ClientCaptureEvent event1 =
+      CreateInternedStringCaptureEvent(kAnswerKey, kAnswerString);
+  orbit_grpc_protos::ClientCaptureEvent event2 =
+      CreateInternedStringCaptureEvent(kNotAnAnswerKey, kNotAnAnswerString);
+
+  auto write_result = output_stream->WriteCaptureEvent(event1);
+  ASSERT_FALSE(write_result.has_error()) << write_result.error().message();
+  write_result = output_stream->WriteCaptureEvent(event2);
+  ASSERT_FALSE(write_result.has_error()) << write_result.error().message();
+  auto close_result = output_stream->Close();
+  ASSERT_FALSE(close_result.has_error()) << close_result.error().message();
+
+  auto capture_file_or_error = CaptureFile::OpenForReadWrite(temporary_file.file_path());
+  ASSERT_TRUE(capture_file_or_error.has_value()) << capture_file_or_error.error().message();
+  std::unique_ptr<CaptureFile> capture_file = std::move(capture_file_or_error.value());
+
+  auto capture_section = capture_file->CreateCaptureSectionInputStream();
+
+  {
+    auto event_or_error = capture_section->ReadEvent();
+    ASSERT_TRUE(event_or_error.has_value()) << event_or_error.error().message();
+    ASSERT_EQ(event_or_error.value().event_case(), ClientCaptureEvent::kInternedString);
+    EXPECT_EQ(event_or_error.value().interned_string().key(), kAnswerKey);
+    EXPECT_EQ(event_or_error.value().interned_string().intern(), kAnswerString);
+  }
+
+  {
+    auto event_or_error = capture_section->ReadEvent();
+    ASSERT_TRUE(event_or_error.has_value()) << event_or_error.error().message();
+    ASSERT_EQ(event_or_error.value().event_case(), ClientCaptureEvent::kInternedString);
+    EXPECT_EQ(event_or_error.value().interned_string().key(), kNotAnAnswerKey);
+    EXPECT_EQ(event_or_error.value().interned_string().intern(), kNotAnAnswerString);
+  }
```

Comment:
Maybe call `ReadEvent()` a third time. I would expect an error then

Reply:
I initially had this code but then I realized that:
1. File section can have a padding (filled with 0's) it used to align sections to 8bytes boundary.
2. Reading from stream filled with 0's yields empty (default) message - so this actually succeeds in the case where we have an additional section.
3. We now have CaptureFinished message to serve as terminator of a stream. This message is expected to be used in real world scenarios.

In conclusion - reading past CaptureFinished message is actually undefined even though the stream is bound to the capture section (sometimes it ends in reading default message other times it results in error, it depends if there is a padding in the main section).

And we should probably not test undefined behavior.

Reply:
I agree. Thank you for the explanation! I would like this to be documented somewhere. A comment at `ReadEvent()` in the Impl.h file seems a good idea

Reply:
added a comment to ReadEvent method